src/state.js

```diff
@@ -0,0 +1,16 @@
+import * as F from 'futil-js'
+
+export let defaultState = {
+  path: null,
+  filterOnly: null,
+  lastUpdateTime: null,
+  markedForDeletion: null,
+  markedForUpdate: null,
+  missedUpdate: null,
+  updating: null,
+  hasValue: null,
+  error: null,
+  results: null,
+}
+
+export let setState = (flat, extend) => F.eachIndexed((node, path) => extend(node, defaultState), flat)
```

Comment:
Don’t forget to call type functions if they exist. That adds support for things like default values, etc